app/src/main/java/org/mozilla/fenix/home/recentsyncedtabs/view/RecentSyncedTab.kt

```diff
@@ -0,0 +1,227 @@
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * A recent synced tab card. Will not render if [recentSyncedTabState] is [RecentSyncedTabState.None].
+ *
+ * @param recentSyncedTabState The state of the recent synced tab.
+ * @param onRecentSyncedTabClick Invoked when the user clicks on the recent synced tab.
+ * @param onSyncedTabSeeAllButtonClick Invoked when user clicks on the "See all" button in the synced tab card.
+ */
+@Composable
+fun RecentSyncedTab(
+    recentSyncedTabState: RecentSyncedTabState,
+    onRecentSyncedTabClick: (RecentSyncedTab) -> Unit = {},
+    onSyncedTabSeeAllButtonClick: () -> Unit = {},
+) {
+    if (recentSyncedTabState == RecentSyncedTabState.None) return
+    val recentSyncedTab = when (recentSyncedTabState) {
+        RecentSyncedTabState.None,
+        RecentSyncedTabState.Loading -> null
+        is RecentSyncedTabState.Success -> recentSyncedTabState.tab
+    }
+    RecentSyncedTabItem(
+        recentSyncedTab,
+        onRecentSyncedTabClick,
+        onSyncedTabSeeAllButtonClick,
+    )
+}
+
+/**
+ * A recent synced tab.
+ *
+ * @param tab Optional synced tab. If null, displays placeholders.
+ * @param onRecentSyncedTabClick Invoked when item is clicked.
+ * @param onSeeAllButtonClick Invoked when "See all" button is clicked.
+ */
+@Suppress("LongMethod")
+@Composable
+private fun RecentSyncedTabItem(
+    tab: RecentSyncedTab?,
+    onRecentSyncedTabClick: (RecentSyncedTab) -> Unit,
+    onSeeAllButtonClick: () -> Unit,
+) {
+    Card(
+        modifier = Modifier
+            .fillMaxWidth()
+            .height(180.dp)
+            .clickable { tab?.let { onRecentSyncedTabClick(tab) } },
+        shape = RoundedCornerShape(8.dp),
+        backgroundColor = FirefoxTheme.colors.layer2,
+        elevation = 6.dp
+    ) {
+        Column(modifier = Modifier.padding(16.dp)) {
+            Row(modifier = Modifier.height(IntrinsicSize.Min)) {
+                if (tab == null) {
+                    RecentTabImagePlaceholder()
+                } else {
+                    ThumbnailCard(
+                        url = tab.url,
+                        key = tab.url.hashCode().toString(),
+                        modifier = Modifier
+                            .size(108.dp, 80.dp)
+                            .clip(RoundedCornerShape(8.dp))
+                    )
+                }
+
+                Spacer(modifier = Modifier.width(16.dp))
+
+                Column(
+                    verticalArrangement = Arrangement.SpaceBetween,
+                    modifier = Modifier.fillMaxHeight()
+                ) {
+                    if (tab == null) {
+                        RecentTabTitlePlaceholder()
+                    } else {
+                        PrimaryText(
+                            text = tab.title,
+                            fontSize = 14.sp,
+                            maxLines = 2,
+                            overflow = TextOverflow.Ellipsis,
```

Comment:
Can we flip this so that we follow the order of the parameters in `PrimaryText()`?